helpers.h

```diff
@@ -0,0 +1,15 @@
+#ifndef HELPERS_H
+#define HELPERS_H
+
+//TODO rename this file - macros.h ?
+//TODO put into dir structure somewhere
+
+#define frand() ((float)rand() / RAND_MAX)
+
+#define prob(p) (frand() <= p)
+
+//TODO do we want this?
+//TODO do we want 1..N?
+#define choose(n) (rand() % (n + 1) + 1)
```

Comment:
Maybe. I'd make it 0 indexed because I'm a robot. Can't be called choose though. Modern languages and libraries always use that to mean "choose something randomly from this list" and that isn't what this is which is confusing. Normally this is some variant or random_range or rand_int